src/Compilers/Core/Portable/Operations/IHasArgumentsExpression.cs

```diff
@@ -11,17 +11,15 @@ namespace Microsoft.CodeAnalysis.Semantics
     public interface IHasArgumentsExpression : IOperation
     {
         /// <summary>
-        /// Arguments of the invocation, excluding the instance argument. Arguments are in parameter order,
-        /// and params/ParamArray arguments have been collected into arrays. Default values are supplied for
-        /// optional arguments missing in source.
+        /// Arguments of the invocation, excluding the instance argument. Arguments are in evaluation order.
         /// </summary>
-        ImmutableArray<IArgument> ArgumentsInParameterOrder { get; }
-        /// <summary>
-        /// Find the argument supplied for a given parameter of the target method.
-        /// </summary>
-        /// <param name="parameter">Parameter of the target method.</param>
-        /// <returns>Argument corresponding to the parameter.</returns>
-        IArgument GetArgumentMatchingParameter(IParameterSymbol parameter);
+        /// <remarks>
+        /// If the invocation is in its expanded form, then params/ParamArray arguments would be collected into arrays. 
+        /// Default values are supplied for optional arguments missing in source. Because they are always constant, their
+        /// evaluation will not impact the evaluation order of the remaining arguments, therefore are appended at the end 
+        /// of the returned argument list.
```

Comment:
Yes, this doesn't sound accurate for VB. I think we should simply remove the "Because they are always constant ..." statement and ensure that implementation uses the order that match the emitted code.